subsys/bluetooth/mesh/sensor_srv.c

```diff
@@ -310,12 +333,44 @@ static int handle_series_get(struct bt_mesh_model *model, struct bt_mesh_msg_ctx
 	bt_mesh_model_msg_init(&rsp, BT_MESH_SENSOR_OP_SERIES_STATUS);
 	net_buf_simple_add_le16(&rsp, id);
 
-	if (!sensor) {
+	if (!sensor || !sensor->series.get) {
+		BT_WARN("No series support in 0x%04x", sensor->type->id);
+		goto respond;
+	}
+
+	if (sensor->type->channel_count < 3) {
+		uint16_t start = 0;
+		uint16_t end = sensor->series.column_count - 1;
+
+		if (buf->len == sizeof(uint16_t) * 2) {
+			start = net_buf_simple_pull_le16(buf);
+			end = net_buf_simple_pull_le16(buf);
+			if (start >= sensor->series.column_count ||
+			    end >= sensor->series.column_count ||
+			    start < end) {
```

Comment:
```suggestion
			    end < start) {
```
?

Reply:
Got me a little dizzy aswell. Is it related to this?
![image](https://user-images.githubusercontent.com/35562897/169847446-43e8fe89-3ed5-4cd1-aa6f-6abdc194a354.png)

Reply:
If I don't miss anything, the for-loop below expects `start` to be before `end`.

Reply:
Then comes my follow-up Q: Is idx 0 equal to the most recent column?